### ISSUE TITLE
Prisma server is crashing when querying more than one pgRelation at once

### ISSUE THREAD

**Issue by berstend** [prisma/prisma#2925]
**Describe the bug**
As mentioned in the slack channel my Prisma server throws Java exceptions when querying multiple relations with a data model derived through `introspect`. 😄 

**To Reproduce**
I've extracted a self-contained test case with instructions here: 
https://github.com/berstend/prisma-relations-bug

![image](https://user-images.githubusercontent.com/1368633/44160103-a6a49d00-a0b9-11e8-99c3-d1c62922b976.png)

```graphql
query {
  titles {
    id
    rating {
      isPopular
    }
    meta {
      releaseYear
    }
  }
}
# querying meta and rating at once will break things
```

**Expected behavior**
* Ideally everything would just work as expected (both `rating` and `meta` return data with no exceptions thrown).
* Neither `prisma introspect` or `prisma deploy` did complain about the schema

**Versions (please complete the following information):**
- I tried with prisma cli / server versions: `1.13`, `1.14-beta`, `1.15-alpha`
- `1.12.3` would not use my `public` schema but always the `management` one.

**Additional context**
I don't think this problem is related to an existing issue on GH, but I'm not entirely sure.

Thanks!

**Comment by marktani** [comment]
Thanks a lot for the amazing bug report @berstend!

I just confirmed it on `1.14.0`.

**Comment by Jaszkowic** [comment]
I have an issue that sounds similar:

This works:

```
query templates(where:
  task: {
    taskGroup: {
      id: "abc"
    }
  }
){
  id
}
```

But this doesn't:

```
query templates(where:
  task: {
    taskGroup: {
      session: {
        id: "xyz"
      }
    }
  }
){
  id
}
```

I don't know if this is the same issue, I will try to reproduce it with a minimal example.

**Comment by marktani** [comment]
Thanks @Jaszkowic, it does look like a separate issue to me! 🙂

**Comment by michalochman** [comment]
I experience similar error (I even think that it's the same).

@berstend can you confirm that if you check the logs, do you see something like the below?

```
java.util.concurrent.ExecutionException: Boxed Error
...
Caused by: java.lang.Error: Passed deferreds should not belong to different relations and should not have different arguments.
```

----

@marktani I was trying something and then I discovered that it works when I try to execute query similar to the following one (note the extra `(where: { id_not: 0 })` in relations):

```graphql
query {
  titles {
    id
    rating(where: { id_not: 0 }) {
      isPopular
    }
    meta(where: { id_not: 0 }) {
      releaseYear
    }
  }
}
```

**Comment by mavilein** [comment]
Thanks for reporting and providing thorough descriptions! 🙏  We fixed this issue and it is available in version `1.15.2` 🎉 . The build will be available in a few minutes.
